### PR TITLE
Add all known `target_feature` configs to check-cfg

### PR DESCRIPTION
**Pull request by Urgau**
This PR adds all the known `target_feature` from ~~`rustc_codegen_ssa`~~ `rustc_target` to the well known list of check-cfg.

It does so by moving the list from `rustc_codegen_ssa` to `rustc_target` ~~`rustc_session` (I not sure about this, but some of the moved function take a `Session`)~~, then using it the `fill_well_known` function.

This already proved to be useful since portable-simd had a bad cfg.

cc @nnethercote (since we discussed it in https://github.com/rust-lang/rust/pull/118494)

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]
Looks super cool, thanks!

**Comment by workingjubilee** [comment]
Is there any chance the target features can be somewhere like `rustc_target`?

**Comment by Urgau** [comment]
> Is there any chance the target features can be somewhere like `rustc_target`?

It should be possible if we are willing to publicly expose the individual lists or remove the `Session` parameter and use a `&str` instead.

Basically everywhere we can have a dependency on `rustc_span` should be possible implementation-wise, since it's quite possible that we will need to make the target features pre-interned symbols for perf reason.

File: compiler/rustc_session/src/target_features.rs [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+        _ => &[],
+    }
+}
+
+pub fn tied_target_features(sess: &crate::Session) -> &'static [&'static [&'static str]] {
+    match &*sess.target.arch {
+        "aarch64" => AARCH64_TIED_FEATURES,
+        _ => &[],
+    }
+}
```
Comment (workingjubilee) [inline compiler/rustc_session/src/target_features.rs:427]: These taking a Session is not particularly intrinsic to their functionality. What matters is that we're burning a path to `sess.target.arch`, which means we're already going through `rustc_target::Target`. These can be turned into functions implemented on `Target`.

Comment (Urgau) [reply compiler/rustc_session/src/target_features.rs]: Done. This is now much cleaner. Thanks you.

**Review by workingjubilee — APPROVED** [review]
With that, the portable-simd change looks good to me. Thanks!

**Comment by TaKO8Ki** [comment]
Thanks! @bors r+

**Comment by Urgau** [comment]
> The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/7209374458/job/19640198548) [(plain)](https://github.com/rust-lang-ci/rust/commit/b1dc6c05f516e81efb33953d9339d72d73d9cd16/checks/19640198548/logs)

Fixed the conflicts with https://github.com/rust-lang/rust/pull/118213. Should be ready for re-approval.

@rustbot ready

**Pull request merged**